### ISSUE TITLE
Allow overwriting image name from the commandline with metadata

### ISSUE THREAD

**Issue by lgeiger** [google-github-actions/deploy-cloudrun#154]
### TL;DR

When working with [metadata customizations](https://github.com/google-github-actions/deploy-cloudrun#metadata-customizations) it is required to set the `image` name in the YAML file.
This can cause problems since it is not easily possible to uniquely tag the image with the git commit hash as recommended in the docs.

It would be great if the image name could be overwritten via the command line even if a metadata YAML file is used

### Design

It would be great if in the following action the image name of the yaml file would be overwritten instead of silently ignored which can cause quite some frustration since only a hard to miss warning is printed in the logs.

**Action YAML**
```yaml
- name: Deploy to Cloud Run
  uses: google-github-actions/deploy-cloudrun@main
  with:
    image: gcr.io/${{ env.PROJECT_ID }}/${{ env.SERVICE }}:${{  github.sha }}
    metadata: ./service.yaml
```

My current ugly workaround for this is:
```yaml
- name: Set Image Name
  run: sed -i 's!REPLACE_IMAGE_NAME!gcr.io/${{ env.PROJECT_ID }}/${{ env.SERVICE }}:${{  github.sha }}!g' service.yaml

- name: Deploy to Cloud Run
  uses: google-github-actions/deploy-cloudrun@main
  with:
    metadata: ./service.yaml
```

**Comment by sethvargo** [comment]
We've updated the action to emit a warning when inputs combinations are invalid. Right now, we don't do any parsing or manipulation of the YAML, and doing so would be a fairly significant refactor. I was going to suggest the `sed` approach, which is what we do in our integration tests 😄
